panoramapublic/src/org/labkey/panoramapublic/query/ExperimentAnnotationsManager.java

```diff
@@ -543,6 +543,20 @@ public static boolean hasProteomicData(ExperimentAnnotations experimentAnnotatio
         return new SqlSelector(PanoramaPublicManager.getSchema(), sql).getObject(Integer.class);
     }
 
+    /**
+     * @param experimentAnnotations
+     * @return true if the given experiment is a journal copy, and is the most recent version.
+     */
+    public static boolean isCurrentVersion(ExperimentAnnotations experimentAnnotations)
+    {
+        if (experimentAnnotations != null && experimentAnnotations.isJournalCopy())
+        {
+            Integer maxVersion = ExperimentAnnotationsManager.getMaxVersionForExperiment(experimentAnnotations.getSourceExperimentId());
+            return experimentAnnotations.getDataVersion().equals(maxVersion);
```

Comment:
Is `getDataVersion()` guaranteed to be non-null? If so, should it return an `int` instead of `Integer`?

Reply:
It is non-null only for experiments that are copies in the Panorama Public project (isJournalCopy()).